esp/tools/soapplus/http.cpp

```diff
@@ -1342,6 +1415,31 @@ int HttpClient::validate(StringBuffer& xml)
         targetns.append(ensptr - nsptr,nsptr);
     }
 
+    char *newxml = strndup(bptr, len);
```

Comment:
You don't need to copy just to add a null terminator.  There is a version of createPTreeFromXMLString that takes a length instead of null terminated strings.

`IPropertyTree *createPTreeFromXMLString(unsigned len, const char *xml, byte flags, PTreeReaderOptions readFlags, IPTreeMaker *iMaker)`